quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnameV2PropertyMappers.java

```diff
@@ -35,13 +40,50 @@ public static PropertyMapper<?>[] getHostnamePropertyMappers() {
         .map(b -> b.isEnabled(() -> Profile.isFeatureEnabled(Profile.Feature.HOSTNAME_V2), "hostname:v2 feature is enabled").build())
         .toArray(s -> new PropertyMapper<?>[s]);
     }
-    
+
     public static void validateConfig() {
         List<String> inUse = REMOVED_OPTIONS.stream().filter(s -> Configuration.getOptionalKcValue(s).isPresent()).toList();
-        
+
         if (!inUse.isEmpty()) {
             LOGGER.errorf("Hostname v1 options %s are still in use, please review your configuration", inUse);
         }
+
+        String host = Configuration.getConfigValue(HttpOptions.HTTP_HOST).getValue();
+        String proxyHeaders = Configuration.getConfigValue(ProxyOptions.PROXY_HEADERS).getValue();
+        if (host == null) {
+            // hostname-strict=false, generally proxy-headers should be set, but it's not required
+            return;
+        }
+        if (proxyHeaders != null) {
+            // must not be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        if (HttpPropertyMappers.isHttpsEnabled()) {
+            // must be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        // if https is not enabled, and any part of the host does not match the rest of the system, then there's likely a problem
+        try {
+            URL url = new URL(host);
+            if ("https".equalsIgnoreCase(url.getProtocol())) {
+                // WARNING, or even ERROR
+                // cors check, nor secure context detection will work as expected
+            }
+            String port = String.valueOf(url.getPort() == -1 ? 80 : url.getPort());
+            String configuredPort = Configuration.getConfigValue(HttpOptions.HTTP_PORT).getValue();
+            if (!port.equals(configuredPort)) {
+                // WARNING
+                // cors check won't work as expected
+            }
+            String path = url.getPath();
+            String configuredPath = Configuration.getConfigValue(HttpOptions.HTTP_RELATIVE_PATH).getValue();
+            // TODO: normalize paths
+            if (!Objects.equals(path, configuredPath)) {
+                // WARNING
+                // cors check won't work as expected
+            }
+        } catch (MalformedURLException e) {
+        }
```

Comment:
IMHO it is fine-ish to keep this validation here (instead of having it in the provider). The providers are encapsulated, they should not have access to config in general, only to their narrow scope.